### PR TITLE
Added a timer on the front end

### PR DESCRIPTION
**Pull request by JamieZafar**

### CODE REVIEW

File: src/client/game/answer-input.js [hunk 1]
```diff
@@ -9,10 +9,17 @@ export default class AnswerInput extends Component {
     componentWillMount() {
         this.setState({
             answerList: [''],
-            focusIndex: 0
+            focusIndex: 0,
+            answerInputTimer: undefined,
```
Comment (kenneth-gray) [inline src/client/game/answer-input.js]: null?

File: src/client/game/answer-input.js [hunk 2]
```diff
@@ -9,10 +9,17 @@ export default class AnswerInput extends Component {
...
+            timerValue: undefined
         });
     }
 
+    componentDidUpdate() {
+        let focusIndex = this.state.focusIndex;
```
Comment (kenneth-gray) [inline src/client/game/answer-input.js]: No need to have this as a separate variable as it's used only one

File: src/client/game/answer-input.js [hunk 3]
```diff
@@ -103,15 +110,43 @@ export default class AnswerInput extends Component {
         );
     }
 
-    componentDidUpdate(){
-        let focusIndex = this.state.focusIndex;
-        this.refs[focusIndex].focus();
+    timerTick() {
+        console.log(this.state);
```
Comment (kenneth-gray) [inline src/client/game/answer-input.js]: Left in a console log?

File: src/client/game/answer-input.js [hunk 4]
```diff
@@ -103,15 +110,43 @@ export default class AnswerInput extends Component {
...
+        if(this.state.timerValue > 0){
+            this.setState({
+                timerValue: this.state.timerValue - 1
+            });
+        }
+        else {
+            clearTimeout(this.state.answerInputTimer);
+            this.setState({
+                answerInputTimer: undefined,
```
Comment (kenneth-gray) [inline src/client/game/answer-input.js]: null?

File: src/client/game/answer-input.js (hunk 1 above)
Comment (JamieZafar) [inline src/client/game/answer-input.js]: we've changed the timerValue to null as we don't do any checks on undefined for it, however we check whether answerInputTimer is undefined later in the code. Should we change it to null to be consistent?

File: src/client/game/answer-input.js [hunk 5]
```diff
@@ -103,15 +110,43 @@ export default class AnswerInput extends Component {
...
+                timerValue: undefined
+            });
+        }
+    }
+
+    startTimer() {
+        let timer = setInterval(this.timerTick.bind(this), 1000);
```
Comment (kenneth-gray) [inline src/client/game/answer-input.js]: Only used once, so please move

File: src/client/game/answer-input.js [hunk 6]
```diff
@@ -103,15 +110,43 @@ export default class AnswerInput extends Component {
...
     }
 
     render() {
+        const timerArea = (
+            <div>
+                <p>Answering Time Left: {this.state.timerValue}</p>
+            </div>
+        )
+        if(this.props.answering && this.state.answerInputTimer === undefined) {
+            this.startTimer();
```
Comment (kenneth-gray) [inline src/client/game/answer-input.js]: This looks super scary - particularly from inside the component render?

File: src/client/game/answer-input.js [hunk 7]
```diff
@@ -103,15 +110,43 @@ export default class AnswerInput extends Component {
...
+                answerInputTimer: undefined,
+                timerValue: undefined
+            });
+        }
+    }
+
+    startTimer() {
+        let timer = setInterval(this.timerTick.bind(this), 1000);
+        this.setState({
+            answerInputTimer: timer,
```
Comment (kenneth-gray) [inline src/client/game/answer-input.js]: I'm not convinced this should be part of the state...

**Comment by kenneth-gray** [comment]
:shipit:

File: src/client/game/answer-input.js [hunk 8]
```diff
@@ -103,15 +122,38 @@ export default class AnswerInput extends Component {
...
+                timerValue: null
+            });
+        }
+    }
+
+
+    startTimer() {
+        this.answerInputInterval = setInterval(() => this.timerTick(), 1000);
+        this.setState({
+            timerValue: 30
```
Comment (dwybourn) [inline src/client/game/answer-input.js]: Best to have this value in a properties file or some javascript equivalent.

**Comment by JamieZafar** [comment]
:sheep: ?

**Pull request merged**